lib/view_component/base.rb

```diff
@@ -173,7 +178,14 @@ def helpers
         )
       end
 
-      @__vc_helpers ||= controller.view_context
+      # Attempt to re-use the original view_context passed to the first
+      # component rendered in the rendering pipeline. This prevents the
+      # instantiation of a new view_context via `controller.view_context` which
+      # always returns a new instance of the view context class.
+
+      # This allows ivars to remain persisted when using the same helper via
+      # `helpers` across multiple components and partials.
+      @__vc_helpers ||= original_view_context || controller.view_context
```

Comment:
We looked at upstreaming a change to `rails/rails` that would memoize `controller.view_context` but the intention seemed to be that a single instance should be instantiated and threaded throughout the render chain. I'm curious if anyone has more context or thoughts about attempting to upstream that to resolve the issue with the intention of eventually reverting this change.